### PR TITLE
refactor(core): remove direct accesses to DOM

### PR DESCRIPTION
**Pull request by marclaval**
Prerequisite for #503

### CODE REVIEW

File: modules/angular2/src/core/compiler/pipeline/view_splitter.js [hunk 1]
```diff
@@ -81,15 +81,15 @@ export class ViewSplitter extends CompileStep {
...
         }
       }
     }
   }
 
   _moveChildNodes(source, target) {
-    while (isPresent(source.firstChild)) {
-      DOM.appendChild(target, source.firstChild);
+    while (isPresent(DOM.firstChild(source))) {
+      DOM.appendChild(target, DOM.firstChild(source));
```
Comment (gkalpak) [inline modules/angular2/src/core/compiler/pipeline/view_splitter.js]: I wonder if JS engines can optimize these identical calls to `DOM.firstChild()`.
Would it make any difference to DRY this up ?

Comment (mlaval) [inline modules/angular2/src/core/compiler/pipeline/view_splitter.js]: Let's DRY anyway :)

File: modules/angular2/src/test_lib/test_lib.es6 [hunk 2]
```diff
@@ -151,12 +151,12 @@ export class SpyObject {
 
 function elementText(n) {
   var hasShadowRoot = (n) => n instanceof Element && n.shadowRoot;
-  var hasNodes = (n) => n.childNodes && n.childNodes.length > 0;
+  var hasNodes = (n) => DOM.childNodes(n) && DOM.childNodes(n).length > 0;
```
Comment (gkalpak) [inline modules/angular2/src/test_lib/test_lib.es6]: You might want to DRY this one up, as well :)

File: modules/angular2/src/test_lib/benchmark_util.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ export function getStringParameter(name:string) {
 
   for (var i=0; i<els.length; i++) {
     el = els[i];
-    if ((el.type !== 'radio' && el.type !== 'checkbox') || el.checked) {
-      value = el.value;
+    if ((DOM.type(el) !== 'radio' && DOM.type(el) !== 'checkbox') || DOM.getChecked(el)) {
```
Comment (gkalpak) [inline modules/angular2/src/test_lib/benchmark_util.js]: This is DRYable too (not sure if it's worth it though).

**Pull request closed without merging**